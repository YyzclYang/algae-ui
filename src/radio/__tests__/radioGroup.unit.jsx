import React from 'react';
import * as renderer from 'react-test-renderer';
import Radio, { RadioGroup } from '../index';
// import { mount } from 'enzyme';

describe('RadioGroup', () => {
  it('渲染一个 RadioGroup', () => {
    const component = renderer
      .create(
        <RadioGroup>
          <Radio>radio1</Radio>
          <Radio>radio2</Radio>
        </RadioGroup>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('根据 options 来渲染 RadioGroup', () => {
    const options = [
      'a',
      { label: 'b', value: 'b' },
      { label: 'c', value: 'c', disabled: true }
    ];
    const component = renderer
      .create(<RadioGroup options={options} />)
      .toJSON();
    const component2 = renderer
      .create(<RadioGroup disabled options={options} />)
      .toJSON();
    expect(component).toMatchSnapshot();
    expect(component2).toMatchSnapshot();
  });

  it('接受 name disabled', () => {
    const component = renderer
      .create(
        <RadioGroup name="test" disabled>
          <Radio>radio1</Radio>
          <Radio>radio2</Radio>
        </RadioGroup>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('children 为 string', () => {
    const component = renderer.create(<RadioGroup>radio</RadioGroup>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 value', () => {
    const component = renderer
      .create(
        <RadioGroup value="1">
          <Radio value="1">radio1</Radio>
          <Radio value="2">radio2</Radio>
        </RadioGroup>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('配置模式接受 value', () => {
    const options = [
      'a',
      { label: 'b', value: 'b' },
      { label: 'c', value: 'c', disabled: true }
    ];
    const component = renderer
      .create(<RadioGroup value="b" options={options} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
