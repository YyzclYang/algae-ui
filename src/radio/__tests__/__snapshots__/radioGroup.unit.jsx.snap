// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup children 为 string 1`] = `
<div
  className="algae-ui-radio-group"
>
  radio
</div>
`;

exports[`RadioGroup 接受 name disabled 1`] = `
<div
  className="algae-ui-radio-group algae-ui-radio-group-disabled"
>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        name="test"
        onChange={[Function]}
        type="radio"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio1
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        name="test"
        onChange={[Function]}
        type="radio"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio2
    </span>
  </label>
</div>
`;

exports[`RadioGroup 接受 value 1`] = `
<div
  className="algae-ui-radio-group"
>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-checked"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={true}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio1
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio2
    </span>
  </label>
</div>
`;

exports[`RadioGroup 根据 options 来渲染 RadioGroup 1`] = `
<div
  className="algae-ui-radio-group"
>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="a"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      a
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="b"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      b
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        onChange={[Function]}
        type="radio"
        value="c"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      c
    </span>
  </label>
</div>
`;

exports[`RadioGroup 根据 options 来渲染 RadioGroup 2`] = `
<div
  className="algae-ui-radio-group algae-ui-radio-group-disabled"
>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        onChange={[Function]}
        type="radio"
        value="a"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      a
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        onChange={[Function]}
        type="radio"
        value="b"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      b
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        onChange={[Function]}
        type="radio"
        value="c"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      c
    </span>
  </label>
</div>
`;

exports[`RadioGroup 渲染一个 RadioGroup 1`] = `
<div
  className="algae-ui-radio-group"
>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio1
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      radio2
    </span>
  </label>
</div>
`;

exports[`RadioGroup 配置模式接受 value 1`] = `
<div
  className="algae-ui-radio-group"
>
  <label
    className="algae-ui-radio-wrapper"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="a"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      a
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-checked"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={true}
        className="algae-ui-radio-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="b"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      b
    </span>
  </label>
  <label
    className="algae-ui-radio-wrapper algae-ui-radio-disabled"
  >
    <span
      className="algae-ui-radio"
    >
      <input
        checked={false}
        className="algae-ui-radio-input"
        disabled={true}
        onChange={[Function]}
        type="radio"
        value="c"
      />
      <span
        className="algae-ui-radio-inner"
      />
    </span>
    <span
      className="algae-ui-radio-text"
    >
      c
    </span>
  </label>
</div>
`;
